### PR TITLE
Adding archive workflow

### PR DESCRIPTION
**Pull request by MariamFahmy98**

### CODE REVIEW

**Comment by MariamFahmy98** [comment]
It is still under progress. There is a small issue in compression when making a backup, I will work on it.

**Comment by MariamFahmy98** [comment]
PTAL

File: CMakeLists.txt [hunk 1]
```diff
@@ -17,6 +17,9 @@ endif(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
 
 project(pgmoneta VERSION ${VERSION_STRING} LANGUAGES  C)
 
+set (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+
```
Comment (jesperpedersen) [inline CMakeLists.txt:22]: Those are in `src/CMakeLists.txt` for `clang`

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (MariamFahmy98) [reply CMakeLists.txt]: I have removed it, Thanks.

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**